### PR TITLE
GumGum: Override the default currency

### PR DESCRIPTION
**Pull request by MartinGumGum**
This PR resolves an issue where bid currency from the Ad Exchange was not being set in the Prebid Server's Go version, causing the currency to default to USD in all cases. The bid currency will now be dynamically set based on the response from the Ad Exchange, ensuring proper handling of different currencies in the bidding process.

### CODE REVIEW

**Comment by MartinGumGum** [comment]
@onkarvhanumante

**Comment by MartinGumGum** [comment]
@minaguib @dlackty @dmitris @asweeney86

**Comment by MartinGumGum** [comment]
@hhhjort Could you please review this PR?

**Comment by MartinGumGum** [comment]
@bsardo  Could you please review this PR?

File: adapters/gumgum/gumgum_test.go [hunk 1]
```diff
@@ -18,3 +22,59 @@ func TestJsonSamples(t *testing.T) {
...
+	bidResponse, errs := bidder.MakeBids(prebidRequest, nil, responseData)
+
+	// Assert no errors
+	if len(errs) != 0 {
+		t.Fatalf("Failed to make bids %v", errs)
+	}
+
+	// Assert that the currency is correctly set
+	assert.Equal(t, expectedCurrency, bidResponse.Currency, "Expected currency to be set to "+expectedCurrency)
+}
```
Comment (bsardo) [inline adapters/gumgum/gumgum_test.go:80]: The guidance is to test your adapter using the JSON test framework, in which case you can get rid of all of this and instead just copy your `exemplary/banner.json` test into the `supplemental` directory, rename it to `convert-currency.json` and then just add `"cur": "EUR"` to `mockResponse.body` and `"currency": "EUR"` to the object in the `expectedBidResponses` array.

Comment (MartinGumGum) [reply adapters/gumgum/gumgum_test.go]: @bsardo  I have addressed the above comment. Could you please review it?  and thx for the review.

**Review by przemkaczmarek — APPROVED** [review]

**Comment by MartinGumGum** [comment]
@bsardo Could you please review this PR again? Thank you!

**Review by bsardo — APPROVED** [review]

**Pull request merged**